webapp/channels/src/components/advanced_text_editor/advanced_text_editor.tsx

```diff
@@ -427,16 +426,25 @@ const AdvancedTextEditor = ({
         };
     }, [handleDraftChange, draft]);
 
-    // Set the draft from store when changing post or channels, and store the previus one
+    // Keep track of the draft as a ref so that we can save it when changing channels
+    const draftRef = useRef(draftFromStore);
     useEffect(() => {
-        setDraft(draftFromStore);
-        return () => handleDraftChange(previousDraft.current, {instant: true, show: true});
-    }, [channelId, postId]);
+        draftRef.current = draft;
+    }, [draft]);
 
-    // Keep track of the previous draft
+    // Set the draft from store when changing post or channels, and store the previous one
     useEffect(() => {
-        previousDraft.current = draft;
-    }, [draft]);
+        // Store the draft that existed when we opened the channel to know if it should be saved
+        const draftOnOpen = draftFromStore;
+
+        setDraft(draftOnOpen);
+
+        return () => {
+            if (draftOnOpen !== draftRef.current) {
```

Comment:
Here we are comparing references. So potentially two empty drafts or two drafts with the same values are considered different. Is that expected for this PR?

Reply:
Yeah, that's intentional. The objects will be the same unless a user makes any changes to the draft, and even if they make changes and then revert them, we might have saved a modified version of the draft that now needs to be reverted. We could potentially track that more granularly, but it seems hard to do without more significant changes